### PR TITLE
Fix toolbar alignment in widget block editor

### LINKED ISSUE #31021: Widgets editor: Toolbar is not left-aligned with the selected block
## Description

The block toolbar in the Widgets screen is not left-aligned with the selected block.

## Step-by-step reproduction instructions
1. Go to Appearance → Widgets.
2. Select any top level block in any widget area.

## Expected behaviour
Block toolbar should be left-aligned with the selected block.

## Actual behaviour
Block toolbar is indented.

## Screenshots or screen recording (optional)
Screen Shot 2021-04-21 at 11 59 19

### PR DESCRIPTION
**Pull request by donnapep**
## Description
Fixes https://github.com/WordPress/gutenberg/issues/31021.

Don't apply the `has-parent` CSS class to the toolbar if the `__experimentalParentSelector` flag is set. This class was applying the left margin to the toolbar when a block was selected in the widget area.

## How has this been tested?

1. In _Appearance_ > _Widgets_, select a top level block in a widget area and ensure toolbar alignment is correct.
2. Add the Navigation block to a page.
3. Click _Add all pages_.
4. Select the inner Page List block and ensure the `block-editor-block-contextual-toolbar` element has the `has-parent` class, and that the toolbar alignment is correct.

## Screenshots 
Screen Shot 2021-05-11 at 7 33 52 AM

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-editor/src/components/block-tools/block-contextual-toolbar.js [hunk 1]
```diff
@@ -18,22 +18,35 @@ import BlockToolbar from '../block-toolbar';
...
+			const parents = getBlockParents( selectedBlockClientId );
+			const firstParentClientId = parents[ parents.length - 1 ];
+			const parentBlockName = getBlockName( firstParentClientId );
+			const parentBlockType = getBlockType( parentBlockName );
+
+			return {
+				blockType:
+					selectedBlockClientId &&
+					getBlockType( getBlockName( selectedBlockClientId ) ),
+				hasParents: getBlockParents( selectedBlockClientId ).length,
```
Comment (noisysocks) [inline packages/block-editor/src/components/block-tools/block-contextual-toolbar.js]:
```suggestion
				hasParents: parents.length,
```

File: packages/block-editor/src/components/block-tools/block-contextual-toolbar.js [hunk 2]
```diff
@@ -18,22 +18,35 @@ import BlockToolbar from '../block-toolbar';
...
+			const {
+				getBlockName,
+				getBlockParents,
+				getSelectedBlockClientIds,
+			} = select( blockEditorStore );
+			const { getBlockType } = select( blocksStore );
+			const selectedBlockClientIds = getSelectedBlockClientIds();
+			const selectedBlockClientId = selectedBlockClientIds[ 0 ];
+			const parents = getBlockParents( selectedBlockClientId );
+			const firstParentClientId = parents[ parents.length - 1 ];
```
Comment (noisysocks) [inline packages/block-editor/src/components/block-tools/block-contextual-toolbar.js:32]: Can `parents` ever be empty?

**Review by noisysocks — COMMENTED** [review]
Thanks for working on this! It works really well when I test it locally.

**Comment by noisysocks** [comment]
Hey @donnapep, could you please rebase `trunk` into this branch?

**Review by noisysocks — APPROVED** [review]

**Review by Procorners — APPROVED** [review]

**Pull request merged**